core/src/main/java/org/apache/accumulo/core/logging/TabletLogger.java

```diff
@@ -0,0 +1,134 @@
+   * loading.
+   */
+  public static void loading(KeyExtent extent, Ample.TServer server) {
+    locLog.debug("Loading {} on {}", extent, server);
+  }
+
+  public static void suspended(KeyExtent extent, HostAndPort server, long time, TimeUnit timeUnit,
+      int numWalogs) {
+    locLog.debug("Suspended {} to {} at {} ms with {} walogs", extent, server,
+        timeUnit.toMillis(time), numWalogs);
+  }
+
+  public static void unsuspended(KeyExtent extent) {
+    locLog.debug("Unsuspended " + extent);
+  }
+
+  public static void loaded(KeyExtent extent, Ample.TServer server) {
+    locLog.debug("Loaded {} on {}", extent, server);
+  }
+
+  public static void unassigned(KeyExtent extent, int logCount) {
+    locLog.debug("Unassigned {} with {} walogs", extent, logCount);
+  }
+
+  public static void split(KeyExtent parent, KeyExtent lowChild, KeyExtent highChild,
+      Ample.TServer server) {
+    locLog.debug("Split {} into {} and {} on {}", parent, lowChild, highChild, server);
+  }
+
+  /**
+   * Called when a tablet's current assignment state does not match the goal state.
+   */
+  public static void missassigned(KeyExtent extent, String goalState, String currentState,
+      Ample.TServer future, Ample.TServer current, int walogs) {
+    // usually this is only called when the states are not equal, but for the root tablet this
+    // method is currently always called
+    if (!goalState.equals(currentState)) {
+      locLog.trace("Miss-assigned {} goal:{} current:{} future:{} location:{} walogs:{}", extent,
+          goalState, currentState, future, current, walogs);
+    }
+  }
+
+  public static void compacted(KeyExtent extent, Collection<? extends Ample.FileMeta> inputs,
+      Ample.FileMeta output) {
+    fileLog.debug("Compacted {} created {} from {}", extent, output, inputs);
+  }
+
+  public static void flushed(KeyExtent extent, Ample.FileMeta absMergeFile,
+      Ample.FileMeta newDatafile) {
+    if (absMergeFile == null)
+      fileLog.debug("Flushed {} created {} from [memory]", extent, newDatafile);
+    else
+      fileLog.debug("Flushed {} created {} from [memory,{}]", extent, newDatafile, absMergeFile);
+  }
+
+  public static void bulkImported(KeyExtent extent, Ample.FileMeta file) {
+    fileLog.debug("Imported {} {}  ", extent, file);
+  }
+
+  public static void recovering(KeyExtent extent, List<LogEntry> logEntries) {
+    if (recoveryLog.isDebugEnabled()) {
+      List<String> logIds = logEntries.stream().map(LogEntry::getUniqueID).collect(toList());
+      recoveryLog.debug("For {} recovering data from walogs: {}", extent, logIds);
+    }
+  }
+
+  public static void recovered(KeyExtent extent, List<LogEntry> logEntries, long numMutation,
+      long numEntries) {
+    recoveryLog.info("For {} recovered {} mutations creating {} entries from {} walogs", extent,
+        numMutation, numEntries, logEntries.size());
+  }
+
+  public static boolean isWalRefLoggingEnabled() {
```

Comment:
That's certainly possible with just the one use of the method in this PR. If more complex checks were needed, you might still want to wrap the logger state check with a more descriptive/logical state check, so you don't have to leak the logger outside this class.

If such a check were necessary in other, more complex cases than what is in this PR, you might be able to do something with a lambda to avoid unnecessary work if trace logging is disabled.

Reply:
I was trying to avoid an expensive computation if tracing was disabled.  I did try using a lambda, but it made the code more complex.  I went with this because I thought it was easier to read.